pkg/api/v20160330/validate.go

```diff
@@ -65,6 +65,9 @@ func (l *LinuxProfile) Validate() error {
 
 // Validate implements APIObject
 func (a *Properties) Validate() error {
+	if a == nil {
```

Comment:
Can you point to some references that suggest that this is idiomatic golang? I think we prefer to do this nil-type validation before the method is invoked.

Reply:
Oh yes, I did think of that initially. But I decided against it as I supposed this it'd be more consistent to put it within the Validate method. I'll make the change and push it.